### PR TITLE
Add support for xip.io service

### PR DESCRIPTION
**Pull request by JPry**
- [ ] @markkelnar
- [ ] @ericmann

VVV [recently](https://github.com/Varying-Vagrant-Vagrants/VVV/pull/449) added support for the [Xip.io](http://xip.io) service. This is extremely useful when attempting to access local development sites from other devices on the same network.

To test this change, do the following:

* Check out the `xipio` branch from [my repo](http://github.com/JPry/hgv).
* Create a file located in `~/.vagrant.d/Vagrantfile` with the following content:
```
# -*- mode: ruby -*-
# vi: set ft=ruby :

Vagrant.configure(2) do |config|
  # Set up with a Public network
  config.vm.network :public_network
end
```
* Run `vagrant reload --provision`
* SSH to the vagrant, and determine the public IP address using `ifconfig`. My experience indicates that the `eth1` interface normally has the public IP, but your results may vary.
* Connect to one of the default sites, adding `.<ip address>.xip.io` to the end of the domain name. In my testing, my IP was `192.168.1.22`, so the domain ended up being `php.hgv.dev.192.168.1.22.xip.io`.

If all goes well, you should see a WordPress site and be able to login and click around normally.

### CODE REVIEW

File: provisioning/roles/listener/templates/etc/nginx/wordpress.conf [hunk 1]
```diff
@@ -1,6 +1,6 @@
 server {
 	listen          80;
-	server_name     {% for i in domain %}{{ i }} {% endfor %};
+	server_name     {% for i in domain %}{{ i }} {{ "~^%s\d+\.\d+\.\d+\.\d+\.xip\.io$"|format(i|replace(".dev",".")|replace(".","\.")) }} {% endfor %};
```
Comment (zamoose) [inline provisioning/roles/listener/templates/etc/nginx/wordpress.conf]: Since we're moving to `.test` eventually and people might want to use `.org`, `.net`, or *shudder* `.co.uk`, `.com.au`, etc., this approach won't work very well. We need to have some sort of TLD-sensitive `replace()` going on there.

Comment (JPry) [inline provisioning/roles/listener/templates/etc/nginx/wordpress.conf]: In that case, maybe it would simply be easiest to not do any TLD replacement at all. It's not really required to do so for xip.io to work properly.

Comment (zamoose) [inline provisioning/roles/listener/templates/etc/nginx/wordpress.conf]: How then does Nginx grok which virtual host to direct connections to?

--
Doug Stewart

Comment (JPry) [inline provisioning/roles/listener/templates/etc/nginx/wordpress.conf]: I changed it to use the full domain name. It is added to each of the domains for a given site. For example, with domains names of `php.hgv.dev` and `php.hgv.test`, the full list of domains that Nginx will accept is:

* `php.hgv.dev`
* `~^php\.hgv\.dev\d+\.\d+\.\d+\.\d+\.xip\.io$`
* `php.hgv.test`
* `~^php\.hgv\.test\d+\.\d+\.\d+\.\d+\.xip\.io$`

Comment (zamoose) [inline provisioning/roles/listener/templates/etc/nginx/wordpress.conf]: Ahh, missed the updated commit. Yeah, that looks good. The `\d` patterns will match any IP address octets, correct? 

I think there may be a missing `\.` between the `%s` and the first `\d`...

`nslookup php.hgv.dev10.0.0.1.xip.io` results in

```
winston:~ zamoose$ nslookup php.hgv.dev10.0.1.1.xip.io
Server:		10.0.0.1
Address:	10.0.0.1#53

Non-authoritative answer:
Name:	php.hgv.dev10.0.1.1.xip.io
Address: 1.0.0.0

winston:~ zamoose$
```

**Comment by JPry** [comment]
@zamoose I just updated with the missing `\.` in the pattern. Good catch.

File: provisioning/roles/listener/templates/etc/nginx/wordpress.conf (hunk 1 above)
Comment (JPry) [inline provisioning/roles/listener/templates/etc/nginx/wordpress.conf]: > The \d patterns will match any IP address octets, correct?

Yep, that's correct.

**Comment by zamoose** [comment]
I don't see any real downsides to this. I'd proceed with looking to merge, @markkelnar / @ericmann / @stephen-lin.

**Comment by ericmann** [comment]
I'm not sure I understand why this is necessary (and had I seen the discussion on VVV when it was added, would have raised my concerns there, too). We already add development domains to the system hostsfile, so the need for dynamic (and very ugly, if you ask me) domains with an IP address in them doesn't feel necessary.

Supporting something like xip.io also means tying HGV to a remote service outside of our control (that may or may not continue to exist/function/etc). The value in running a local emulator of a hosted system (i.e. Mercury) is that you can dev without network access - requiring a remote DNS provider like xip.io removes that ability.

TL;DR; I'm not sold on us needing this right now.

**Comment by ericmann** [comment]
Closing this for now (please see my above comment for reasoning). That being said, we can easily re-open things in the future if we decide to move forward with xip.io support.

**Comment by videoMonkey** [comment]
@ericmann I just want to know what method you recommend that I might be able to access my vagrant site on my computer from my mobile phone testing purposes? I like xip.io because it's simple and it works, but I'm open to suggestions. I just need to be able to test my theme development from actual devices and other virtual machines

**Comment by iamntz** [comment]
>  We already add development domains to the system hosts file

Right, but these domains will work only from the host computer, right? How would you test the site on a mobile physical device?

Up to a point you can develop by resizing browser & co, but at a certain moment you'll need to switch to real device testing _before_ launching the site/app :)
